src/app/cli/src/coda_worker_testnet.ml

```diff
@@ -80,11 +80,31 @@ module Api = struct
   let teardown t = Deferred.List.iter t.workers ~f:Coda_process.disconnect
 end
 
+module Tree = struct
+  type key = string
+
+  type t = {parent_map: (key, key) Hashtbl.t}
+
+  let add t ~prev ~curr = Hashtbl.add t.parent_map ~key:curr ~data:prev
+
+  let create () = {parent_map= Hashtbl.create (module String)}
+
+  let path_from t node =
+    let rec go acc cur =
+      match Hashtbl.find t.parent_map cur with
+      | Some parent -> go (parent :: acc) parent
+      | None -> List.rev acc
+    in
+    go [] node
+end
+
 let start_prefix_check log workers events testnet ~acceptable_delay =
   let all_transitions_r, all_transitions_w = Linear_pipe.create () in
-  let chains = Array.init (List.length workers) ~f:(fun i -> []) in
+  let state_hash_tree = Tree.create () in
+  let chains = Array.init (List.length workers) ~f:(fun i -> "") in
   let check_chains chains =
     let chains = Array.filteri chains ~f:(fun i _ -> Api.synced testnet i) in
+    let chains = Array.map chains ~f:(Tree.path_from state_hash_tree) in
```

Comment:
Is this a chain anymore?

Reply:
they _are_ chains now, they weren't before.

Reply:
you're right, the names could use some work.